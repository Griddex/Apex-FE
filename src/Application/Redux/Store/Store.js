import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./../Reducers/RootReducer";
import createSagaMiddleware from "redux-saga";
import watchLoginSaga from "../Sagas/LoginSaga";
import watchNewProjectSaga from "../../../Project/Redux/Sagas/NewProjectSaga";
import watchRegisterSaga from "../Sagas/RegisterSaga";
import watchConnectDatabaseSaga from "../../../Import/Redux/Sagas/connectDatabaseSaga";
import { spawn } from "redux-saga/effects";
import authMiddleware from "./../Middlewares/AuthMiddleware";
import uiSpinnerMiddleware from "./../Middlewares/UISpinnerMiddleware";
import watchRunForecastSaga from "../../../Network/Redux/Sagas/RunForecastSaga";

function* rootSaga() {
  yield spawn(watchLoginSaga);
  yield spawn(watchRegisterSaga);
  yield spawn(watchConnectDatabaseSaga);
  yield spawn(watchRunForecastSaga);
  yield spawn(watchNewProjectSaga);
}

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(uiSpinnerMiddleware, authMiddleware, sagaMiddleware)
  )
);

sagaMiddleware.run(rootSaga);

export default store;
