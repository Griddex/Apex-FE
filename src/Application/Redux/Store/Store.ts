import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import allReducers from "../Reducers/AllReducers";
import createSagaMiddleware from "redux-saga";
import watchLoginSaga from "../Sagas/LoginSaga";
import watchCreateNewProjectSaga from "../../../Project/Redux/Sagas/CreateNewProjectSaga";
import watchRegisterSaga from "../../../Administration/Redux/Sagas/RegisterSaga";
import watchConnectDatabaseSaga from "../../../Import/Redux/Sagas/ConnectDatabaseSaga";
import { spawn } from "redux-saga/effects";
import authMiddleware from "../Middlewares/AuthMiddleware";
import uiSpinnerMiddleware from "../Middlewares/UISpinnerMiddleware";
import watchRunForecastSaga from "../../../Network/Redux/Sagas/RunForecastSaga";
import rootReducer from "../Reducers/RootReducer";
import watchFetchRecentProjectsSaga from "../../../Project/Redux/Sagas/FetchRecentProjectsSaga";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import watchOpenRecentProjectSaga from "../../../Project/Redux/Sagas/OpenRecentProjectSaga";
import watchFetchExistingDataSaga from "../../../Import/Redux/Sagas/FetchExistingDataSaga";
import watchSaveInputDeckSaga from "../../../Import/Redux/Sagas/SaveInputDeckSaga";
import watchAutogenerateNetworkSaga from "../../../Network/Redux/Sagas/AutogenerateNetworkSaga";
import watchSaveForecastParametersSaga from "../../../Network/Redux/Sagas/SaveForecastParametersSaga";
import watchSaveNetworkSaga from "../../../Network/Redux/Sagas/SaveNetworkSaga";
import watchFetchUnitSettingsSaga from "../../../Settings/Redux/Sagas/Sagas/UnitSettingsSaga";
import watchFetchApplicationHeadersSaga from "../../../Import/Redux/Sagas/FetchApplicationHeadersSaga";
import watchAndSaveAutogenerateNetworkSaga from "../../../Network/Redux/Sagas/SaveAndAutogenerateNetworkSaga";
import watchFetchExistingForecastParametersSaga from "../../../Network/Redux/Sagas/FetchExistingForecastParametersSaga";
import watchFetchExistingNetworkDataSaga from "../../../Network/Redux/Sagas/FetchExistingNetworkDataSaga";
import watchGenerateNetworkBySelectionSaga from "../../../Network/Redux/Sagas/GenerateNetworkBySelectionSaga";
import watchUpdateForecastParametersSaga from "../../../Network/Redux/Sagas/UpdateForecastParametersSaga";
import watchSaveForecastSaga from "../../../Network/Redux/Sagas/SaveForecastSaga";
import watchGetForecastResultsSaga from "../../../Forecast/Redux/Sagas/GetForecastResultsSaga";
import watchFetchUserDetailsSaga from "../Sagas/FetchUserDetailsSaga";
import watchFetchExistingForecastingResultsSaga from "../../../Forecast/Redux/Sagas/FetchExistingForecastingResultsSaga";
import watchFetchTreeviewKeysSaga from "../../../Forecast/Redux/Sagas/FetchTreeviewKeysSaga";

function* rootSaga() {
  yield spawn(watchLoginSaga);
  yield spawn(watchRegisterSaga);
  yield spawn(watchConnectDatabaseSaga);
  yield spawn(watchCreateNewProjectSaga);
  yield spawn(watchFetchRecentProjectsSaga);
  yield spawn(watchFetchUnitSettingsSaga);
  yield spawn(watchOpenRecentProjectSaga);
  yield spawn(watchFetchExistingDataSaga);
  yield spawn(watchSaveInputDeckSaga);
  yield spawn(watchAutogenerateNetworkSaga);
  yield spawn(watchAndSaveAutogenerateNetworkSaga);
  yield spawn(watchRunForecastSaga);
  yield spawn(watchSaveForecastParametersSaga);
  yield spawn(watchUpdateForecastParametersSaga);
  yield spawn(watchSaveNetworkSaga);
  yield spawn(watchFetchApplicationHeadersSaga);
  yield spawn(watchFetchExistingNetworkDataSaga);
  yield spawn(watchFetchExistingForecastParametersSaga);
  yield spawn(watchGenerateNetworkBySelectionSaga);
  yield spawn(watchSaveForecastSaga);
  yield spawn(watchGetForecastResultsSaga);
  yield spawn(watchFetchUserDetailsSaga);
  yield spawn(watchFetchExistingForecastingResultsSaga);
  yield spawn(watchFetchTreeviewKeysSaga);
}

// function listenMongoDb() {
//   try {
//     mongoose.connect(
//       "mongodb+srv://apex:Yv10NPk7c2WmgKt4@syncware-cluster.a69fi.mongodb.net/apex-db?retryWrites=true&w=majority"
//     );

//     const db = mongoose.connection;
//     const taskCollection = db.collection("datastreams");
//     const changeStream = taskCollection.watch();
//     changeStream.on("change", (change) => {
//       console.log("Changes made to DataStream");
//     });
//   } catch (err) {
//     console.log(err);
//   }
// }

function listenWebsocket() {
  // const socket = new WebSocket()
}

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  // configuration object for redux-persist
  key: "root",
  storage, // define which storage to use
};
// const persistedRootReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(uiSpinnerMiddleware, authMiddleware, sagaMiddleware)
  )
);

// const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);

// export { store, persistor };
// listenMongoDb();
export { store };
