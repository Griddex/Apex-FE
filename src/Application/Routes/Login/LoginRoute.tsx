import { Grid, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import React from "react";
import { LoginForm } from "../../Components/Forms/LoginForm";
import ApexLogoFull from "../../Images/ApexLogoFull.svg";
import Image from "../../Components/Visuals/Image";

const useStyles = makeStyles(() => ({
  root: {
    flexWrap: "nowrap",
    height: "100%",
    textAlign: "center",
  },
  item: {
    height: "100%",
    marginTop: "20vh",
  },
  image: {
    marginBottom: "40px",
    height: "15vmin",
  },
  typography: {
    color: "#808080",
    margin: "20px 20px 30px 20px",
    width: 500,
  },
}));

const LoginRoute = () => {
  const classes = useStyles();

  return (
    <Grid
      container
      className={classes.root}
      direction="column"
      justify="center"
      alignItems="center"
    >
      <Grid item className={classes.item} xs={10} sm={6} lg={4}>
        <Image
          className={classes.image}
          src={ApexLogoFull}
          alt="Hydrocarbon Forecasting Platform Company Logo"
        />
        <Typography className={classes.typography} variant="h5">
          - Robust Hydrocarbon Forecasting and Economics Evaluation Platform
        </Typography>
        <LoginForm />
      </Grid>
    </Grid>
  );
};

export default LoginRoute;
