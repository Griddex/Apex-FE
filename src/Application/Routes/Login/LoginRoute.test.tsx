import React from "react";
import { render, waitFor, cleanup } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { CssBaseline } from "@material-ui/core";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";
import { ThemeProvider } from "@material-ui/core";
import { Provider } from "react-redux";
import DateFnsUtils from "@date-io/date-fns";

import LoginRoute from "./LoginRoute";
import theme from "../../Theme/Theme";
import { store } from "../../Redux/Store/Store";

const history = createMemoryHistory();
const Component = (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <Router history={history}>
          <CssBaseline />
          <LoginRoute />
        </Router>
      </MuiPickersUtilsProvider>
    </ThemeProvider>
  </Provider>
);
afterEach(cleanup);

describe("Tests the root component <App/>", () => {
  test("renders without errors", async () => {
    const { getByText } = render(Component);
    const mottoElement = await waitFor(() =>
      getByText(
        /- Versatile Hydrocarbon Forecasting and Economics Evaluation Platform/i
      )
    );
    expect(mottoElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(Component);
    expect(container).toMatchSnapshot();
  });

  test("Apex logo renders", async () => {
    const { findByAltText } = render(Component);
    const logoElement = await findByAltText(
      "Hydrocarbon Forecasting Platform Company Logo"
    );
    expect(logoElement).toBeInTheDocument();
  });

  test("Login is successful with registered user", () => {});
});
