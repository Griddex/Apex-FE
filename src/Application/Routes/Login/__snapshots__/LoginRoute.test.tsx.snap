// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the root component <App/> matches snapshot 1`] = `
<div>
  <div
    style="display: flex; flex-direction: row; align-items: center; justify-content: center; width: 100%; height: 100%;"
  >
    <div
      class="makeStyles-root-13"
      style="min-width: 500px; width: 70%; max-width: 600px; height: auto;"
    >
      <img
        alt="Hydrocarbon Forecasting Platform Company Logo"
        class="makeStyles-image-14"
        src="ApexLogoFull.svg"
      />
      <h6
        class="MuiTypography-root makeStyles-typography-15 MuiTypography-h6"
      >
        - Versatile Hydrocarbon Forecasting and Economics Evaluation Platform
      </h6>
      <form
        action="#"
        class="makeStyles-form-16"
      >
        <div
          class="makeStyles-formTextFields-17"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
            >
              Username
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                name="userName"
                required=""
                type="text"
                value=""
              />
            </div>
            
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required"
              data-shrink="true"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
                name="password"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            
          </div>
        </div>
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-24"
                data-indeterminate="false"
                type="checkbox"
                value="remember"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Remember me
          </span>
        </label>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-19 MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
